### PR TITLE
Add scripts to keyword field mapper

### PR DESCRIPTION
**Pull request by romseygeek**
This commit adds `script` and `on_script_error` parameters to 
keyword field mappers, allowing you to define index-time scripts
for keyword fields.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java [hunk 1]
```diff
@@ -90,15 +95,26 @@ private static KeywordFieldMapper toType(FieldMapper in) {
...
+
         private final IndexAnalyzers indexAnalyzers;
+        private final ScriptCompiler scriptCompiler;
 
-        public Builder(String name, IndexAnalyzers indexAnalyzers) {
+        public Builder(String name, IndexAnalyzers indexAnalyzers, ScriptCompiler scriptCompiler) {
             super(name);
             this.indexAnalyzers = indexAnalyzers;
+            this.scriptCompiler = scriptCompiler;
+            this.script.precludesParameters(nullValue);
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:108]: how about ignore_above and normalizer? Should they be supported when a script is set?

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java [hunk 2]
```diff
@@ -90,15 +95,26 @@ private static KeywordFieldMapper toType(FieldMapper in) {
...
+            this.script.setValidator(s -> {
+                if (s != null && indexed.get() == false && hasDocValues.get() == false) {
+                    throw new MapperParsingException("Cannot define script on field with index:false and doc_values:false");
+                }
+            });
         }
 
         public Builder(String name) {
-            this(name, null);
+            this(name, null, null);
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:117]: ehm, I think you'll have to use ScriptCompiler.NONE here

**Review by javanna — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java (hunk 1 above)
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java:108]: Yes, I've included those - they're both things that are applied after a value has been read, as opposed to `null_value` which is applied *instead* of a value.

**Pull request merged**